app/routes/ad.tsx

```diff
@@ -1,10 +1,31 @@
 import { MetaFunction } from "@remix-run/node";
+import { useSearchParams } from "@remix-run/react";
 
 import { AdGrid } from "~/components/AdGrid";
 import { createMeta } from "~/config";
 
 export const meta: MetaFunction = () => createMeta({ page: "Ad" });
 
+function useYearAndMonth() {
+	const [searchParams] = useSearchParams();
+
+	const currentDate = new Date();
```

Comment:
I know that's not exactly what the linked issue suggests, but how do you feel about using `YYYY-MM` instead?

The URL could be `?date=2024-09`. Then you can pass it directly to:

```tsx
const paramDate = searchParams.get("month");
const date = paramDate ? new Date(paramDate) : new Date();
```

if `new Date(paramDate)` is not valid, don't display anything on the page.

Then you can get the correct string with:

```ts
date.toLocaleDateString(undefined, { month: "long", year: "numeric" });
```

(TBD: I'm not sure if `locale` should be `"default"` or `undefined`, if this should use `toLocaleDateString` or `toLocaleString`, etc.)